### PR TITLE
varnishd: Warn when workdir is not  on tmpfs

### PR DESCRIPTION
**Pull request by walid-git**
Adds a warning when workdir is not on a tmpfs mounted partition.
Refs #4121

### CODE REVIEW

File: bin/varnishd/mgt/mgt_jail.c [hunk 1]
```diff
@@ -177,7 +182,12 @@ VJ_make_workdir(const char *dname)
 		    "can not reside on a file system mounted noexec\n", dname);
 	}
 #endif
+#if defined(HAVE_FSTATFS) && defined(TMPFS_MAGIC)
+	struct statfs sfs[1];
 
+	if (!fstatfs(i, sfs) && sfs->f_type != TMPFS_MAGIC)
+			MGT_Complain(C_WARN, "Workdir is not mounted on tmpfs\n");
```
Comment (dridi) [inline bin/varnishd/mgt/mgt_jail.c:189]: Double indent.

**Review by dridi — APPROVED** [review]

**Comment by dridi** [comment]
bugwash: make a new linux jail to isolate linuxisms like this one and make it reuse the bits from the unix jail.

**Comment by nigoroll** [comment]
I can see the motivation of the bugwash decision, but I feel like it is complicating things for no relevant gain. Today, the linux jail would be identical to the unix jail except for ~2 lines added, and we could still (easily) start the linux jail once more specifics arise.
I think this addition has its merits already.

**Comment by bsdphk** [comment]
The linux jail can just call into the standard unix-jail for everything which does not differ.

The reason I want this kind of platform-magic segregated is that other wise we end up with a nightmare of #ifdefs all over the place.

**Comment by dridi** [comment]
> The linux jail can just call into the standard unix-jail for everything which does not differ.

This is how we are approaching this. It should be "that simple" initially, and we should actually be able to grow new linux-specific containment features. While I sympathize with @nigoroll's goal to get this in now, and take care of the linux jail later, I also sympathize with @bsdphk's view that if let this go in, we lose an incentive to actually materialize the linux jail and centralize linux-specific checks.

Luckily for us, @cartoush is already working on it. There's also another linux-specific piece of code in the unix jail, and it will find a new home in the process.

I think we can close it for good.

**Pull request closed without merging**